### DISCUSSION TITLE
kong idle timeout increase from 60s

### DISCUSSION THREAD

**Discussion by prasad-asokan** [Kong/kong#12112] [Help] [answered]
Tried using multiple approaches and none of them seem to work. We are using KIC and Kong gateway and by default, idle timeout is 60s. We have tried keepalive_timeout, proxy_read_timeout and nothing is working. This is not working as well. 

KONG_UPSTREAM_KEEPALIVE_IDLE_TIMEOUT

Can someone help if this is configurable? We use NLB and I understand it has keep alive_timeout as 350s so that shouldn't be an issue.

**Comment by chobits** [comment, marked as answer]
> KONG_UPSTREAM_KEEPALIVE_IDLE_TIMEOUT
>
> Can someone help if this is configurable? We use NLB and I understand it has keep alive_timeout as 350s so that shouldn't be an issue.

Since the version 2.7.0, this option has been renamed from nginx_upstream_keepalive_timeout to upstream_keepalive_idle_timeout. And you can configure it in kong.conf.

You could also find it here https://github.com/Kong/kong/blob/master/kong.conf.default#L1029
